### PR TITLE
C++: Improve StdSet and StdPair models

### PR DESCRIPTION
**Pull request by geoffw0**
Improve `StdSet.qll` and `StdPair.qll` models:
 - modelling the class instantiations as well as class member functions (this has been found to be a little bit cleaner, easier to work with and more efficient)
 - `StdSet.qll` now includes bsl compatibility (bsl is missing `insert_or_assign` and `merge` but that should be harmless)
 - `StdPair.qll` now includes bsl compatibility (though the bsl has some extra stuff, such as the base classes for pair, that we don't model; this could be improved but should be harmless)

### CODE REVIEW

File: cpp/ql/src/semmle/code/cpp/models/implementations/StdPair.qll [hunk 1]
```diff
@@ -7,8 +7,8 @@ import semmle.code.cpp.models.interfaces.Taint
 /**
  * An instantiation of `std::pair<T1, T2>`.
  */
-class StdPairClass extends ClassTemplateInstantiation {
-  StdPairClass() { getTemplate().hasQualifiedName("std", "pair") }
+class StdPair extends ClassTemplateInstantiation {
```
Comment (MathiasVP) [inline cpp/ql/src/semmle/code/cpp/models/implementations/StdPair.qll:10]: This one isn't `private`. Do we need to add a `deprecated` alias?

**Review by MathiasVP — APPROVED** [review]
LGTM otherwise!

File: cpp/ql/src/semmle/code/cpp/models/implementations/StdPair.qll (hunk 1 above)
Comment (geoffw0) [reply cpp/ql/src/semmle/code/cpp/models/implementations/StdPair.qll:10]: Yeah - `StdPairClass` hasn't existed for very long but it probably has been in at least one release of some form, so it deserved both of those things.

Comment (geoffw0) [reply cpp/ql/src/semmle/code/cpp/models/implementations/StdPair.qll]: Done.

**Review by MathiasVP — DISMISSED** [review]

**Comment by jbj** [comment]
Why doesn't `StdPairConstructor` get BSL support in this PR?

**Review by criemen — CHANGES_REQUESTED** [review]
LGTM, but I believe `@jbj`s comment needs to be addressed before this can go in.

**Comment by geoffw0** [comment]
Ah well spotted, I had missed `StdPairConstructor`.  Fixed now.

**Review by criemen — APPROVED** [review]

**Pull request merged**